components/encryption/src/manager/mod.rs

```diff
@@ -238,7 +234,7 @@ impl Dicts {
         Ok(())
     }
 
-    fn link_file(&self, src_fname: &str, dst_fname: &str) -> Result<()> {
+    fn link_file(&self, src_fname: &str, dst_fname: &str) -> Result<Option<()>> {
```

Comment:
Why returning `Result<Option<()>>`? Is it possible to replace it with `Result<bool>`?

Reply:
Both represent two states. The problem with using `bool` is knowing what it means. For `is_empty` it is obvious, but in cases like these it is not. The file apis here are being changed to return an `Option` with `None` meaning the operation did not occur because the file did not exist. So `Option<T>` is the consistent choice for these APIs and in this case `Option<()>` will be understandable to someone familiar with the APIs.